Dockerfile

```
@@ -8,6 +8,10 @@ RUN yum -y install epel-release && yum -y update
 
 RUN yum -y install rsync ca-policy-egi-core fetch-crl
 
+RUN curl -LO https://dl.k8s.io/release/v1.29.0/bin/linux/amd64/kubectl
+RUN chmod +x ./kubectl
+RUN mv ./kubectl /usr/local/bin
```

Usually there is only one `RUN` command (if possible):
```suggestion
RUN curl -LO https://dl.k8s.io/release/v1.29.0/bin/linux/amd64/kubectl \
 && chmod +x ./kubectl \
 && mv ./kubectl /usr/local/bin
```
which you can then merge with the following one. `curl` also allows to download directly in `/usr/local/bin/kubectl` (`-o` option).
Said that, have you considered putting these commands directly into `update-trust-anchors.sh`, inside the `if [ -n "${CA_BUNDLE_SECRET_TARGET}" ]; then`? in that way, `kubectl` would be available in the container only if needed; before downloading it, you could check if it's already present, in case the script has already been run.

---

Sorry for the delay. I was waiting until we moved it to almalinux9 because the build was broken.

It makes sense! I changed accordingly.